File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/04_quicksort/julia/04_recursive_max.jl
using Test

function max_(arr)
    if length(arr) == 1
        return arr[1]
    end
    sub_max = max_(arr[2:end])
    return arr[1] > sub_max ? arr[1] : sub_max
end

@test max_([1,5,8,20,9,10]) == 20

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/04_quicksort/kotlin/04_recursive_max.kt
private fun max(list: IntArray): Int = when {
    list.size == 2 -> if (list[0] > list[1]) list[0] else list[1]
    else -> {
        val subMax = max(list.copyOfRange(1, list.size))
        if (list[0] > subMax) list[0] else subMax
    }
}

fun main(args: Array<String>) = println(max(intArrayOf(1, 5, 10, 25, 16, 1))) // 25
